posthog/api/cohort.py

```diff
@@ -647,6 +647,55 @@ def update(self, cohort: Cohort, validated_data: dict, *args: Any, **kwargs: Any
                         "Please remove the cohort from these feature flags before deleting it."
                     )
 
+                # Check if cohort is used in test_account_filters
+                teams_with_cohort = Team.objects.filter(
+                    project_id=cohort.team.project_id, test_account_filters__contains=[{"type": "cohort"}]
+                )
+                teams_using_cohort = []
+                for team in teams_with_cohort:
+                    for filter_item in team.test_account_filters:
+                        if filter_item.get("type") == "cohort" and filter_item.get("value") == cohort.id:
+                            teams_using_cohort.append(team)
+                            break
+
+                if teams_using_cohort:
+                    team_names = [team.name for team in teams_using_cohort]
+                    raise ValidationError(
+                        f"This cohort is used in 'Filter out internal and test users' for {len(teams_using_cohort)} environment(s): {', '.join(team_names)}. "
+                        "Please remove the cohort from these test account filters before deleting it."
+                    )
+
+                # Check if cohort is used in insights
+
+                from posthog.models.insight import Insight
```

Comment:
Minor nit: This should be imported at the top with all the other imports unless there's a good reason to import it here (such as to prevent a circular import).